src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -127,56 +114,75 @@ private void RunBlazorStartupScripts(AWebView view)
 			// Set up JS ports
 			view.EvaluateJavascript(@"
 
-        const channel = new MessageChannel();
-        var nativeJsPortOne = channel.port1;
-        var nativeJsPortTwo = channel.port2;
-        window.addEventListener('message', function (event) {
-            if (event.data != 'capturePort') {
-                nativeJsPortOne.postMessage(event.data)
-            }
-            else if (event.data == 'capturePort') {
-                if (event.ports[0] != null) {
-                    nativeJsPortTwo = event.ports[0]
-                }
-            }
-        }, false);
-
-        nativeJsPortOne.addEventListener('message', function (event) {
-        }, false);
-
-        nativeJsPortTwo.addEventListener('message', function (event) {
-            // data from native code to JS
-            if (window.external.__callback) {
-                window.external.__callback(event.data);
-            }
-        }, false);
-        nativeJsPortOne.start();
-        nativeJsPortTwo.start();
-
-        window.external.sendMessage = function (message) {
-            // data from JS to native code
-            nativeJsPortTwo.postMessage(message);
-        };
-
-        window.external.receiveMessage = function (callback) {
-            window.external.__callback = callback;
-        }
-
-        ", new JavaScriptValueCallback(() =>
+		const channel = new MessageChannel();
+		var nativeJsPortOne = channel.port1;
+		var nativeJsPortTwo = channel.port2;
+		window.addEventListener('message', function (event) {
+			if (event.data != 'capturePort') {
+				nativeJsPortOne.postMessage(event.data)
+			}
+			else if (event.data == 'capturePort') {
+				if (event.ports[0] != null) {
+					nativeJsPortTwo = event.ports[0]
+				}
+			}
+		}, false);
+
+		nativeJsPortOne.addEventListener('message', function (event) {
+		}, false);
+
+		nativeJsPortTwo.addEventListener('message', function (event) {
+			// data from native code to JS
+			if (window.external.__callback) {
+				window.external.__callback(event.data);
+			}
+		}, false);
+		nativeJsPortOne.start();
+		nativeJsPortTwo.start();
+
+		window.external.sendMessage = function (message) {
+			// data from JS to native code
+			nativeJsPortTwo.postMessage(message);
+		};
+
+		window.external.receiveMessage = function (callback) {
+			window.external.__callback = callback;
+		}
+
+		", new JavaScriptValueCallback(() =>
 			{
 				// Set up Server ports
 				_webViewHandler?.WebviewManager?.SetUpMessageChannel();
 
 				// Start Blazor
 				view.EvaluateJavascript(@"
-                    Blazor.start();
-                ", new JavaScriptValueCallback(() =>
+					Blazor.start();
+				", new JavaScriptValueCallback(() =>
 				{
 					// Done; no more action required
 				}));
 			}));
 		}
 
+		private static bool IsAppOriginPageUri(string? requestUriString)
+		{
+			if (string.IsNullOrEmpty(requestUriString))
+			{
+				return false;
+			}
+
+			var appBaseUri = new Uri(AppOrigin);
```

Comment:
Consider adding a field like this at the top of the class and then using it in this method:

```c#
		private static readonly Uri AppOriginUri = new Uri(AppOrigin);
```

Because this can't change during app execution.